src/components/o-testimonials/js/testimonials.js

```diff
@@ -140,7 +166,8 @@ class Testimonials extends UiEvents {
   autoRotate() {
     // auto rotate until disabled
     if (!this.autoRotateDisabled) {
-      setTimeout(
+      clearTimeout(this.autoRotateTimer);
+      this.autoRotateTimer = setTimeout(
```

Comment:
you could use `debounce`

Reply:
I realized I don't need debounce here. Added check whether this timer has already been initialized and it works good now.

Reply:
alright